src/libcore/convert.rs

```diff
@@ -30,6 +50,10 @@ use marker::Sized;
 ///
 /// [book]: ../../book/borrow-and-asref.html
 ///
+/// **Note:** these traits are for trivial conversion. **They must not fail**. If
```

Comment:
It may be worth it to call out here that there's no technical reason why they should not fail right now, it's just idiomatic that they should never fail.